chevah/empirical
# [#2699] Make twisted/zope.interface an optional dependecy.

Problem
======

We want to use emprirical for projects which don't  depend on Twisted or zope interface.

Changes
=======

Make twisted and zope optional.

Right now empirical is a mess so I have just patch it up and added more junk... as we slowly move toward consolidating our project as open source project we will see how to change and clean empirical.


How to test
========

reviewers: @alibotean 

check that changes make sense

compat and empirical changes were used in chevah-keycert project https://github.com/chevah/chevah-keycert
which is our new open source project which is tested on Travis-CI and also has coverage reporting and is publicly published on official PyPi index.

Thanks!
